### PR TITLE
qdma4: preconfig for slave bridge and enable ST

### PR DESCRIPTION
**Pull request by karenx-xilinx**
- add config settings for slave bridge (if present)
- enable streaming mode (if present)
- qdma: allow multiple requests (buffers) in one io_submit
- qdma4: pass user irq index instead of actual irq vector to the user interrupt handler.

### CODE REVIEW

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/qdma.c [hunk 1]
```diff
@@ -1250,7 +1250,7 @@ static ssize_t queue_write_iter(struct kiocb *kiocb, struct iov_iter *io)
 	qdma = queue->qdma;
 
 	nr = io->nr_segs;
-	if (!iter_is_iovec(io) || nr != 2) {
+	if (!iter_is_iovec(io) || (nr & 0x1)) {
```
Comment (houlz0507) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/qdma.c:1253]: what is this change for?

Comment (karenx-xilinx) [reply src/runtime_src/core/pcie/driver/linux/xocl/subdev/qdma.c:1253]: Hi, Lizhi. This is to allow multiple requests in 1 io_submit(), each request will take 2 buffers.

**Comment by houlz0507** [comment]
Seems unrelated uboot failure. schedule another run
14:14:26 NOTE: recipe petalinux-image-minimal-1.0-r0: task do_rootfs: Started
14:15:13 ERROR: petalinux-image-minimal-1.0-r0 do_rootfs: The postinstall intercept hook 'update_gtk_immodules_cache' failed, details in /tmp/aarch64-2020.07.20-14.05.58-ak8/work/zynqmp_generic-xilinx-linux/petalinux-image-minimal/1.0-r0/temp/log.do_rootfs
14:15:13 ERROR: Logfile of failure stored in: /tmp/aarch64-2020.07.20-14.05.58-ak8/work/zynqmp_generic-xilinx-linux/petalinux-image-minimal/1.0-r0/temp/log.do_rootfs.123295
14:15:13 NOTE: recipe petalinux-image-minimal-1.0-r0: task do_rootfs: Failed
14:15:13 ERROR: Task (/opt/xrt/build/aarch64/components/yocto/layers/meta-petalinux/recipes-core/images/petalinux-image-minimal.bb:do_rootfs) failed with exit code '1'
14:15:14 NOTE: Tasks Summary: Attempted 6015 tasks of which 5670 didn't need to be rerun and 1 failed.
14:15:14 
14:15:14 Summary: 1 task failed:
14:15:14   /opt/xrt/build/aarch64/components/yocto/layers/meta-petalinux/recipes-core/images/petalinux-image-minimal.bb:do_rootfs
14:15:14 Summary: There was 1 ERROR message shown, returning a non-zero exit code.
14:15:14 ERROR: Failed to build xrt

**Comment by houlz0507** [comment]
retest this please

**Comment by houlz0507** [comment]
retest this please

**Comment by houlz0507** [comment]
retest this please

**Comment by houlz0507** [comment]
retest this please

**Comment by dayeh-xilinx** [comment]
retest this please

**Pull request merged**